identity/src/ecdsa.rs

```diff
@@ -118,10 +126,14 @@ impl SecretKey {
 
     /// Try to parse a secret key from a byte buffer containing raw scalar of the key.
     pub fn try_from_bytes(buf: impl AsRef<[u8]>) -> Result<SecretKey, DecodingError> {
-        SigningKey::from_bytes(buf.as_ref())
+        SigningKey::from_bytes(buf.as_ref().into())
             .map_err(|err| DecodingError::failed_to_parse("ecdsa p256 secret key", err))
             .map(SecretKey)
     }
+
+    pub fn encode_pkcs8_der(&self) -> Vec<u8> {
+        self.0.to_pkcs8_der().unwrap().to_bytes().to_vec()
```

Comment:
Are we sure this `unwrap` can never fail? Can you replace it with an `expect` instead that documents on how the invariant is upheld?